data-serving/data-service/src/model/case.ts

```diff
@@ -1,43 +1,289 @@
 import mongoose from 'mongoose';
+import { ObjectId, Double } from 'mongodb';
 
-export const validOutcomes = ['pending', 'recovered', 'death'];
+export enum Sex {
+    Female,
+    Male,
+    Other,
+}
 
-const demographicsSchema = new mongoose.Schema({
-    age: {
+export enum Outcome {
+    Pending,
+    Recovered,
+    Death,
+}
+
+export enum Species {
+    HomoSapien,
+}
+
+export enum TravelPurpose {
+    Family,
+    Conference,
+    Work,
+}
+
+const dateValidator = {
+    type: Date,
+    min: '2019-11-01',
+    max: Date.now,
+};
+
+const rangeSchema = new mongoose.Schema({
+    minimum: {
+        type: Number,
+        min: -1,
+        max: 200,
+    },
+    maximum: {
         type: Number,
         min: 0,
         max: 200,
     },
+});
+
+const ageSchema = new mongoose.Schema({
+    years: {
+        type: Number,
+        min: -0.75,
+        max: 200,
+    },
+    range: rangeSchema,
+});
+
+const demographicsSchema = new mongoose.Schema({
+    ageSchema,
+    species: {
+        type: String,
+        enum: Object.keys(Species),
+    },
     sex: {
         type: String,
-        enum: ['female', 'male'],
+        enum: Object.keys(Sex),
     },
 });
 
-const caseSchema = new mongoose.Schema({
-    date: {
-        type: Date,
-        min: '2019-11-01',
-        max: Date.now,
-        required: 'Enter a date.',
+const eventSequenceSchema = new mongoose.Schema({
+    onsetSymptoms: dateValidator,
+    confirmed: dateValidator,
+    hospitalAdmission: dateValidator,
+    deathOrDischarge: dateValidator,
+});
+
+const dictionaryValueSchema = new mongoose.Schema({
+    provided: {
+        type: [String],
+        uniqueItems: true,
+    },
+    imputed: {
+        type: [String],
+        uniqueItems: true,
+    },
+    other: {
+        type: [String],
+        uniqueItems: true,
+    },
+});
+
+const metadataSchema = new mongoose.Schema({
+    moderatorId: {
+        type: String,
+        required: 'Enter a moderator id',
+    },
+    submissionDate: dateValidator,
+});
+
+const outbreakSpecificsSchema = new mongoose.Schema({
+    livesInWuhan: Boolean,
+    reportedMarketExposure: Boolean,
+});
+
+const sourceSchema = new mongoose.Schema({
+    id: String,
+    url: String,
+    other: String,
+});
+
+const pathogenSchema = new mongoose.Schema({
+    name: {
+        type: String,
+        required: 'Enter a pathogen name',
+    },
+    source: sourceSchema,
```

Comment:
What does this represent in the context of a pathogen, again?

Reply:
Where the sequence data can be found, I believe. That's coming from this comment: https://docs.google.com/document/d/1fMPiwohsN2vDyeW7TafPLvAUij4bi49KjB4pTV7kJco/edit?disco=AAAAJcTSnIQ

Looking back at it, it looks like this should be named SequenceSource perhaps